File: savestateobserver/src/main/java/com/pnuema/savestateobserver/SaveStateLibraryContentProvider.java
package com.pnuema.savestateobserver;

import android.app.Application;

import com.pnuema.savestateobserver.observers.ActivitySaveStateObserver;

public final class SaveStateLibraryContentProvider extends BaseContentProvider {
    @Override
    public boolean onCreate() {
        if (getContext() instanceof Application) {
            ((Application) getContext()).registerActivityLifecycleCallbacks(new ActivitySaveStateObserver());
        }
        return super.onCreate();
    }
}

File: savestateobserver/src/main/java/com/pnuema/savestateobserver/SaveStateLibraryContentProvider.kt
package com.pnuema.savestateobserver

import android.app.Application

import com.pnuema.savestateobserver.observers.ActivitySaveStateObserver

class SaveStateLibraryContentProvider : BaseContentProvider() {
    override fun onCreate(): Boolean {
        if (context is Application) {
            (context as Application).registerActivityLifecycleCallbacks(ActivitySaveStateObserver())
        }
        return super.onCreate()
    }
}
